.github/workflows/nightly.yml

```diff
@@ -292,6 +299,46 @@ jobs:
           docker rmi test-image
           rm -rf app/tomee
 
+      - name: Run Archetype for EE 9, SE 8, TomEE
+        run: |
+          mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9 -DjavaVersion=8 -Druntime=tomee -DoutputDirectory=app/tomee -Dgoals="clean package" | tee mvn_output.txt
+          
+          MAVEN_EXIT_CODE=${PIPESTATUS[0]}
+          ERROR_MESSAGE="TomEE requires Web Profile and Jakarta EE 8 or 9.1."
+          if ! { [ $MAVEN_EXIT_CODE -ne 0 ] && grep -q "$ERROR_MESSAGE" mvn_output.txt; }; then
+              echo "Maven build did not fail, or the expected error message was not found. Test failed."
+              exit 1
+          fi
+          
+          rm -f mvn_output.txt   
+
+      - name: Run Archetype for EE 9.1, SE 8, TomEE
+        run: |
+          mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9.1 -DjavaVersion=8 -Druntime=tomee -DoutputDirectory=app/tomee -Dgoals="clean package" | tee mvn_output.txt
+          
+          MAVEN_EXIT_CODE=${PIPESTATUS[0]}
+          ERROR_MESSAGE="TomEE requires Web Profile and Jakarta EE 8 or 9.1."
```

Comment:
The error messages you need to check for will be from the Archetype, not the UI. Please look here: https://github.com/eclipse-ee4j/starter/blob/master/archetype/src/main/resources/META-INF/archetype-post-generate.groovy.